Source/src/core/GameObject.cpp

```diff
@@ -699,3 +702,16 @@ void Hachiko::GameObject::ChangeColor(float4 color, float time)
         child->ChangeColor(color, time);
     }
 }
+
+void Hachiko::GameObject::DeleteRemovedComponents()
+{
+    if (!components_to_remove.empty())
+    {
+        for (auto it = components_to_remove.begin(); it != components_to_remove.end(); ++it)
+        {
+            ForceRemoveComponent(*it);
+            delete *it;
+        }
+        components_to_remove.clear();
```

Comment:
For me it would make more sense to store a vector of components to remove in scene manager and not in each individual gameobject. I think doing it in this centralized way could make it cheaper and simpler to understand. Also i see that u use the selected gameobject for that but we could just process the queue of components to be deleted no matter how they ended up there.

Reply:
normally components are not deleted in bulk, although this code does allow for that, how do you say it would work storing them in the scene manager? components are relative to a single game object, we would need to also store the game object pointer, to make sure we remove the component from the game object component's vector

Reply:
Couldnt we just store component pointers? What causes the gameobject pointer to be necessary?

Reply:
In the spirit of centralization, it's possible for the component to be removed, to belong to an object that was just removed, thus leaving us on invalid memory when we try to remove the component

Reply:
changes made according to teacher's feedback